sui-execution/latest/sui-adapter/src/execution_engine.rs

```diff
@@ -220,100 +218,69 @@ fn execute_transaction<Mode: ExecutionMode>(
-        match check_limit_by_meter!(
-            !gas_charger.is_unmetered(),
-            effects_estimated_size,
-            protocol_config.max_serialized_tx_effects_size_bytes(),
-            protocol_config.max_serialized_tx_effects_size_bytes_system_tx(),
-            metrics.excessive_estimated_effects_size
-        ) {
-            LimitThresholdCrossed::None => (),
-            LimitThresholdCrossed::Soft(_, limit) => {
-                warn!(
-                    effects_estimated_size = effects_estimated_size,
-                    soft_limit = limit,
-                    "Estimated transaction effects size crossed soft limit",
-                )
-            }
-            LimitThresholdCrossed::Hard(_, lim) => {
-                execution_result = Err(ExecutionError::new_with_source(
-                    ExecutionErrorKind::EffectsTooLarge {
-                        current_size: effects_estimated_size as u64,
-                        max_size: lim as u64,
-                    },
-                    "Transaction effects are too large",
-                ))
-            }
-        };
         if execution_result.is_ok() {
-            // This limit is only present in Version 3 and up, so use this to gate it
-            if let (Some(normal_lim), Some(system_lim)) = (
-                protocol_config.max_size_written_objects_as_option(),
-                protocol_config.max_size_written_objects_system_tx_as_option(),
-            ) {
-                let written_objects_size = temporary_store.written_objects_size();
-
-                match check_limit_by_meter!(
-                    !gas_charger.is_unmetered(),
-                    written_objects_size,
-                    normal_lim,
-                    system_lim,
-                    metrics.excessive_written_objects_size
-                ) {
-                    LimitThresholdCrossed::None => (),
-                    LimitThresholdCrossed::Soft(_, limit) => {
-                        warn!(
-                            written_objects_size = written_objects_size,
-                            soft_limit = limit,
-                            "Written objects size crossed soft limit",
-                        )
-                    }
-                    LimitThresholdCrossed::Hard(_, lim) => {
-                        execution_result = Err(ExecutionError::new_with_source(
-                            ExecutionErrorKind::WrittenObjectsTooLarge {
-                                current_size: written_objects_size as u64,
-                                max_size: lim as u64,
-                            },
-                            "Written objects size crossed hard limit",
-                        ))
-                    }
-                };
-            }
+            check_written_objects_limit::<Mode>(
+                temporary_store,
+                gas_charger,
+                protocol_config,
+                metrics,
+            )?;
```

Comment:
same here I believe, returning would change the semantic, we want to assign to `execution_result` right?
and am I even reading this right?
there are no tests error with this change?
could you please double check? @oxade?

Reply:
No tests failed, but I have reverted it to not return until the end 👍